R/rules.R

```diff
@@ -525,18 +764,43 @@ sv_between <- function(left,
 #'   echoed along with text that states how many extra elements are part of the
 #'   `set`.
 #'
-#' @return A function suitable for using as an
+#' @return A function suitable for use as an
 #'   [`InputValidator$add_rule()`][InputValidator] rule.
-#'
+#' 
 #' @examples
-#' # Ignore withReactiveDomain(), it's just required to get this example to run
-#' # outside of Shiny
-#' shiny::withReactiveDomain(shiny::MockShinySession$new(), {
+#' ## Only run examples in interactive R sessions
+#' if (interactive()) {
 #'
+#' library(shiny)
+#' library(shinyvalidate)
+#' 
+#' ui <- fluidPage(
+#'   textInput("rating", "Rating")
+#' )
+#' 
+#' server <- function(input, output, session) {
+#'   
+#'   # Validation rules are set in the server, start by
+#'   # making a new instance of an `InputValidator()`
 #'   iv <- InputValidator$new()
-#'
-#'   iv$add_rule("rating", sv_in_set(1:5, set_limit = 5))
-#' })
+#' 
+#'   # Basic usage: `sv_in_set()` requires a value
+#'   # set given as a vector; a message will be
+#'   # shown if the validation of `input$rating` fails
+#'   iv$add_rule("rating", sv_in_set(1:5))
+#' 
+#'   # Finally, `enable()` the validation rules
+#'   iv$enable()
+#' }
+#' 
+#' shinyApp(ui, server)
+#' 
+#' }
+#' 
+#' @family rule functions
+#' 
+#' @seealso The [sv_between()] function, which tests whether a field values
+#'   between two boundary values.
```

Comment:
Can we try testing with passing a factor to `sv_in_set` and then using a string value?